app/uk/gov/hmrc/gform/auth/AuthService.scala

```diff
@@ -224,6 +224,8 @@ class AuthService(
 
       case SubmissionReference => getSubmissionReference(envelopeId).pure[Future]
 
+      case Constant(value) => Future.successful(value)
```

Comment:
Seeing this now in bigger context, we need to completely get rid of this whole method `evaluateSubmissionReference`.
It is duplicating (incompletely) what `eval` on `uk.gov.hmrc.gform.graph.Evaluator` is for. Is there something preventing us to use this `eval` instead?

Reply:
Not that I know of. I wasn't aware that Evaluator even existed. Is removing evaluateSubmissionReference something for me to do or to leave for another ticket ?

Reply:
No need for another ticket, it is work which can safely be done as part of this ticket.

Reply:
How would you deal with the fact that `eval` requires imports that `evaluateSubmissionReference` doesn't? There are things that `eval` needs that we don't have access to where `evaluateSubmissionReference` is called